main/test/flix/Test.Exp.Fixpoint.KeywordSolve.flix

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2025 Casper Dalgaard Nielsen
+ *                Adam Yasser Tallouzi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+mod Test.Exp.Fixpoint.KeywordSolve {
+
+    /////////////////////////////////////////////////////////////////////////////
+    // Test that `solve` does not mutate the original program being solved     //
+    /////////////////////////////////////////////////////////////////////////////
+
+    @Test
+    pub def testSolve01(): Bool = {
+        let solved = solve #{
+            ASolve01(42).
+        };
+        let otherProg = #{
+            ASolve01(-42).
+        };
+        let k = solve solved, otherProg;
+        let res1 = query solved select x from ASolve01(x);
+        let res2 = query k select x from ASolve01(x);
+        res1 `Assert.eq` Vector#{42} and res2 `Assert.eq` Vector#{-42, 42}
+    }
```

Comment:
I think these tests would be easier to read if it was simply:

db1
db2

and/or mm1 and mm2.

That way the text aligns better.

I would also just call the predicate `A`.

Reply:
I see

Reply:
But I think tests are independent, so only within the same file.